local Controllers = script.Parent:WaitForChild("Controllers")

local InputController = require(Controllers:WaitForChild("InputController"))
local MovementController = require(Controllers:WaitForChild("MovementController"))
local CameraController = require(Controllers:WaitForChild("CameraController"))
local HUDController = require(Controllers:WaitForChild("HUDController"))
local CharacterSelectController = require(Controllers:WaitForChild("CharacterSelectController"))
local CombatController = require(Controllers:WaitForChild("CombatController"))
local FeedbackController = require(Controllers:WaitForChild("FeedbackController"))
local MobileControlsController = require(Controllers:WaitForChild("MobileControlsController"))
local ArenaHUDController = require(Controllers:WaitForChild("ArenaHUDController"))
local TeamPanelController = require(Controllers:WaitForChild("TeamPanelController"))

InputController.Start()
CameraController.Start()
HUDController.Start()
FeedbackController.Start()
MovementController.Start()
CharacterSelectController.Start()
CombatController.Start()
MobileControlsController.Start()
ArenaHUDController.Start()
TeamPanelController.Start()

print("[Aniverse Royale] AR-003 client controllers loaded.")
